### PR TITLE
Update Makefile and index.rst for docs

### PR DESCRIPTION
**Pull request by kay-kim**
Update makefile's target [migrate] to copy over everything in api/ directory - rather than keep up with directories under api.

### CODE REVIEW

File: docs/api/Makefile [hunk 1]
```diff
@@ -43,9 +42,9 @@ html: Makefile migrate
 # Also, on macOS, need -i '' but not for Linux
 
 ifdef IS_MAC
-	find $(SOURCECOPYDIR)/modules/ -type f -name "*.md" -exec sed -i '' -e 's?\.md#?\.html#?g' {} \;
+	find $(SOURCECOPYDIR)api/modules/ -type f -name "*.md" -exec sed -i '' -e 's?\.md#?\.html#?g' {} \;
```
Comment (Vinnl) [inline docs/api/Makefile:45]: Given that it used to say `/modules`, shouldn't we prepend a slash to `api/modules` as well? (Same question for the `else` leg.)

```suggestion
	find $(SOURCECOPYDIR)/api/modules/ -type f -name "*.md" -exec sed -i '' -e 's?\.md#?\.html#?g' {} \;
```

**Review by Vinnl — APPROVED** [review]
I'm not sure if I understand what's going on/why this is needed, but it looks fine to me (assuming the answer to the question I asked is "no" :) ).

File: docs/api/Makefile (hunk 1 above)
Comment (kay-kim) [reply docs/api/Makefile:45]: Oh - SOURCECOPYDIR already ends with a / and on a mac, the extra slash of build/source//api  causes it to error.

Of course, I now see that I'm only changing in the api modules/ files and if files in the classes/ directory link to other files, this sed won't change it.

Comment (kay-kim) [reply docs/api/Makefile:45]: But - adding a "." to the "find  . $(SOURCECOPYDIR)"  should allows us to build with the extra "/" .  Am making the change now.

Comment (kay-kim) [reply docs/api/Makefile]: Oh - actually - just realized I clobbered logic I shouldn't have during the copy phase.

**Pull request merged**